packages/legacy-scripting-runner/test/example-app/index.js

```diff
@@ -1171,6 +1171,19 @@ const App = {
       source: "return z.legacyScripting.run(bundle, 'hydrate.file');"
     }
   },
+  // This breaks `applyBeforeMiddleware` which it looks like gets the original definition without `findSourceRequireFunctions` applied?!
+  // beforeRequest: [
+  //   {
+  //     source: 'return z.legacyScripting.beforeRequest(request, z, bundle);',
+  //     args: ['request', 'z', 'bundle']
+  //   }
+  // ],
```

Comment:
@eliangcs it looks like I ran into a bug here. When I enable this, the `hook trigger > `KEY_catch_hook` integration tests fail because the `source` here hasn't been replaced by a function via `findSourceRequireFunctions` when it gets called by `applyBeforeMiddleware`.

Any idea? We don't strictly need `beforeRequest` here (just `afterResponse` - and that works fine), but it may be good to check if this is a test-setup issue or an actual bug in the legacy scripting runner.